shared/mad/codeql/mad/ModelValidation.qll

```diff
@@ -0,0 +1,177 @@
+/**
+ * Provides classes and predicates related to validating models-as-data rows.
+ */
+
+/** Holds if a model exists for the given `kind`. */
+signature predicate modelKindSig(string kind);
+
+/** Provides validation for models-as-data summary, sink, source, and neutral kinds. */
+module KindValidation<
+  modelKindSig/1 summaryKind, modelKindSig/1 sinkKind, modelKindSig/1 sourceKind,
```

Comment:
You could consider to use module as input. This will also means that it is possible to define default predicates (which could be used for the neutralKind) in the signature module. Maybe something like
```codeql
signature module KindValidationConfig {

  /** Holds if a summary model exists for the given `kind`. */
  predicate summaryKind(string kind);

  /** Holds if a sink model exists for the given `kind`. */
  predicate sinkKind(string kind);

  /** Holds if a source model exists for the given `kind`. */
  predicate sourceKind(string kind);

  /** Holds if a neutral model exists for the given `kind`. */
  default predicate neutralKind(string kind) { none() }
}

module KindValidation<KindValidationConfig Config> {
...
}
```

Reply:
Agree. Also, this would guard against accidentally providing the argument predicates in the wrong order as they would be provided by-name rather than by-position since they all have the same signature.

Reply:
Thanks for the suggestion! Updated in https://github.com/github/codeql/pull/13324/commits/9abe3e3da464c3f65f7641e31b2e1a7a06163ac9. Let me know if this needs any additional edits.